timesketch/frontend-ng/src/components/UploadForm.vue

```diff
@@ -0,0 +1,552 @@
+      let i = this.mandatoryHeaders.findIndex((h) => h.name === target)
+      if (lastElementSelected === 'Create New Header') {
+        this.mandatoryHeaders[i].columnsSelected = ['Create New Header']
+      } else {
+        this.mandatoryHeaders[i].columnsSelected = this.mandatoryHeaders[i].columnsSelected.filter(
+          (h) => h !== 'Create New Header'
+        )
+      }
+      let sources = []
+      let defaultValue = null
+      if (lastElementSelected === 'Create New Header') {
+        sources = null
+        do {
+          defaultValue = prompt('Insert the default value for this header')
+          if (defaultValue.includes(this.CSVDelimiter)) {
+            alert(`New header value cannot contain CSV separator (found ${this.CSVDelimiter})`)
+            defaultValue = null
+          }
+        } while (!defaultValue)
+      } else {
+        sources = this.mandatoryHeaders[i].columnsSelected
+      }
+
+      this.headersMapping = this.headersMapping.filter((mapping) => mapping['target'] !== target)
+      if (sources === null || sources.length > 0)
+        this.headersMapping.push({
+          target: target,
+          source: sources,
+          default_value: defaultValue,
+        })
+      this.validateFile()
+    },
+    clearFormData: function () {
+      this.fileMetaData = {}
+      this.form.name = ''
+      this.form.file = ''
+      this.fileName = ''
+      this.headersMapping = []
+      this.infoMessage = ''
+      this.headersString = ''
+      this.valuesString = []
+      this.uploadedFiles = []
+      this.title = 'Upload your Plaso/JSONL/CSV file'
+      this.error = []
+      this.mandatoryHeaders = [
+        { name: 'datetime', columnsSelected: [] },
+        { name: 'message', columnsSelected: [] },
+        { name: 'timestamp_desc', columnsSelected: [] },
+      ]
+    },
+    submitForm: function () {
+      if (!this.validateFile()) {
+        return
+      }
+
+      let formData = new FormData()
+      formData.append('file', this.form.file)
+      formData.append('name', this.form.name)
+      formData.append('provider', 'WebUpload')
+      formData.append('context', this.fileName)
+      formData.append('total_file_size', this.form.file.size)
+      formData.append('sketch_id', this.$store.state.sketch.id)
+      if (['csv', 'jsonl', 'json'].includes(this.extension)) {
+        let hMapping = JSON.stringify(this.headersMapping)
+        formData.append('headersMapping', hMapping)
+        formData.append('delimiter', this.CSVDelimiter)
+      }
+      let config = {
+        headers: {
+          'Content-Type': 'multipart/form-data',
+        },
+        onUploadProgress: function (progressEvent) {
+          this.percentCompleted = Math.round((progressEvent.loaded * 100) / progressEvent.total)
+        }.bind(this),
+      }
+      ApiClient.uploadTimeline(formData, config)
+        .then((response) => {
+          this.$store.dispatch('updateSketch', this.$store.state.sketch.id)
```

Comment:
You can set sketch in computed to have access to the sketch object in this component

```
  computed: {
    sketch() {
      return this.$store.state.sketch
    }
}
```